File: solutions/java/high-scores/1/src/main/java/HighScores.java

import java.util.Comparator;
import java.util.List;

class HighScores {

    private final List<Integer> scores;

    public HighScores(List<Integer> highScores) {
        scores = List.copyOf(highScores);
    }

    List<Integer> scores() {
        return scores;
    }

    Integer latest() {
        return scores.getLast();
    }

    Integer personalBest() {
        return scores.stream()
            .max(Comparator.naturalOrder())
            .orElseThrow();
    }

    List<Integer> personalTopThree() {
        return scores.stream()
            .sorted(Comparator.reverseOrder())
            .limit(3)
            .toList();
    }

}

File: solutions/fsharp/high-scores/1/HighScores.fs
module HighScores

type Scores = int list

let scores (values: Scores) = values

let latest (values: Scores) = List.last values

let personalBest (values: Scores) = List.max values

let personalTopThree (values: Scores): Scores =
    values |> List.sortDescending |> List.truncate 3
